File: practice23/4.cpp

//https://www.nowcoder.com/practice/37aa8a88a72e47f798a14d63bee61d8f?tpId=98&tqId=32851&tPage=2&rp=2&ru=/ta/2019test&qru=/ta/2019test/question-ranking
#include<iostream>
using namespace std;

int main()
{
	int n;
	while (cin >> n)
	{
		if (n == 1 || n == 2 || n == 4)  //1 2 4装不满
		{
			cout << "-1" << endl;
		}
		else
		{
			int count = 0;
			count += n / 7; //对7取余，对余数进行讨论即可
			n = n % 7;
			if (n == 3 || n == 5 || n == 1)
			{
				count += 1;
			}
			if (n == 2 || n == 4 || n == 6)
			{
				count += 2;
			}
			cout << count << endl;
		}
	}

	system("pause");
	return 0;
}

File: practice33/4.java

//https://www.nowcoder.com/practice/6ffdd7e4197c403e88c6a8aa3e7a332a?tpId=85&tqId=29861&tPage=2&rp=2&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.math.BigInteger;
import java.util.*;
import java.util.Scanner;

public class Main {
    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        BigInteger result=new BigInteger("1");
        BigInteger temp;
        for(Integer i=2;i<=n;i++){
            temp=new BigInteger(i.toString());
            result=result.multiply(temp);
        }
        String str=result.toString();
        int count=0;
        for(int i=str.length()-1;i>=0;i--){
            if(str.charAt(i)=='0'){
                count++;
            }
            else{
                break;
            }
        }
        System.out.println(count);
    }
}